### PR TITLE
day 2 solution

### PR DESCRIPTION
**Pull request by pearlbea**
Got part 1, but my answer for part 2 was too big.

### CODE REVIEW

File: day2/day2.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  const reports = data.split(/\n/).map((report) => {
+    return report.split(" ").map((item) => {
+      return parseInt(item, 10);
+    });
+  });
+  return reports;
+}
+
+export function isIncreasing(array: number[]) {
+  for (let i = 0; i < array.length; i++) {
```
Comment (latteier) [inline day2/day2.ts:16]: I'm wondering if these loops are off by one. Seems like you'd need length -1 comparisons? Maybe it doesn't matter?

File: day2/day2.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      // added for part 2
+      safeReports++;
+    }
+  }
+  return safeReports;
+}
+
+console.log(numberOfSafeReports(parseFile(data))); // 356 part 1, 498 (too big) for part 2
+
+export function canBeSafe(report: number[]) {
```
Comment (latteier) [inline day2/day2.ts:63]: I wonder if not getting the right result for part two has something to do with generating the sub-reports incorrectly somehow - maybe worth breaking the code that generates sub-reports out into a function and testing it?

File: day2/day2.test.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    expect(report).toBe(false);
+  });
+
+  it("should return the number of safe reports", () => {
+    const reports = parseFile(data);
+    const safeReports = numberOfSafeReports(reports);
+    expect(safeReports).toBe(2);
+  });
+
+  it("should test saftey with one level removed", () => {
```
Comment (latteier) [inline day2/day2.test.ts:55]: not sure if you can stumble upon an illuminating test case - but here's where you could verify what's going on with part two.

**Review by latteier — APPROVED** [review]
I love your solution! It's straightforward and looks correct to me. I'm not sure why part 2 isn't working.

File: day2/day2.ts (hunk 2 above)
Comment (pearlbea) [reply day2/day2.ts:69]: Yep! Fixed. Still getting the wrong result, but it's a different result 😄

**Comment by latteier** [comment]
maybe we should just merge this and go on to day 3?

**Pull request merged**